### PR TITLE
Remove classes optional typings

### PR DESCRIPTION
**Pull request by Markionium**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Changes `ColorClassNames`  to use `Partial<T>`

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by Markionium** [comment]
@dzearing I think this isn't as useful as hoped though and basically doesn't change much. Because of the dynamic property creation it seems it'll be difficult to type `ColorClassNames` with non-optionals. Even though we know that all the classNames will be there :)

**Comment by Markionium** [comment]
@dzearing I was hoping it wasn't but TS won't allow me to assign `{}` to `ColorClassNames`. Since we know all the props will be there we could do an `as ColorClassNames` but thats not great either. 

```ts
export const ColorClassNames: Partial<IColorClassNames> = {};
```
I tried to "fix" it by reducing instead of building it up with a `for in` loop.

TS is technically doing the correct thing since removing a color from the `DefaultPallete` would result into non-existent properties. (Afaik it isn't possible to have dynamic types based on the keys of an object? It might be that due to my relatively short experience with TS I'm missing something)

I also considered changing the ColorClassNames into something like the following, so we could do something like below, but it leaves me with the same issue around the dynamic properties.

```ts
interface ColorClasses {
    text: string;
    textHover: string;
    background: string;
    backgroundHover: string;
    border: string;
    borderHover: string;
}

export type IColorClassesNames = {[key in keyof IPalette]: ColorClasses};
```
PS: It's not such a big issue as i resolved on our side, but perhaps still worth it to fix it somehow?

File: packages/styling/src/classNames/ColorClassNames.ts [hunk 1]
```diff
@@ -6,297 +6,297 @@ import { DefaultPalette } from '../styles/DefaultPalette';
...
+  greenLight: string;
+  greenLightHover: string;
+  greenLightBackground: string;
+  greenLightBackgroundHover: string;
+  greenLightBorder: string;
+  greenLightBorderHover: string;
 }
 
-export const ColorClassNames: IColorClassNames = {};
+export const ColorClassNames: Partial<IColorClassNames> = {};
```
Comment (dzearing) [inline packages/styling/src/classNames/ColorClassNames.ts]: Shouldn't this be:

```tsx
export const ColorClassNames: IColorClassNames = {} as IColorClassNames;
```

Then in your code, you don't need to treat it as an non-optional?

File: packages/styling/src/classNames/ColorClassNames.ts [hunk 2]
```diff
@@ -324,7 +324,7 @@ for (const colorName in DefaultPalette) {
  * Defines a getter for the given class configuration.
  */
 function _defineGetter(
-  obj: IColorClassNames,
+  obj: Partial<IColorClassNames>,
```
Comment (dzearing) [inline packages/styling/src/classNames/ColorClassNames.ts]: You don't need this if you do what I said above.

**Pull request merged**